centreon/www/class/centreonDB.class.php

```diff
@@ -213,24 +229,462 @@ public function __construct($db = self::LABEL_DB_CONFIGURATION, $retry = 3)
+                throw new CentreonDbException(
+                    "Binding parameters are empty",
+                    ['bind_params' => $bindParams]
+                );
+            }
+
+            if ($withParamType) {
+                foreach ($bindParams as $paramName => $bindParam) {
+                    if (is_array($bindParam) && !empty($bindParam) && count($bindParam) === 2) {
+                        $paramValue = $bindParam[0];
+                        $paramType = $bindParam[1];
+                        if (
+                            !in_array(
+                                $paramType,
+                                [PDO::PARAM_STR, PDO::PARAM_BOOL, PDO::PARAM_INT, PDO::PARAM_NULL],
+                                true
+                            )
+                        ){
+                            throw new CentreonDbException(
+                                "Error for the param type, it's not an integer or a value of PDO::PARAM_*",
+                                ['bind_param' => $bindParam]
+                            );
+                        }
+                        $this->makeBindValue($pdoStatement, $paramName, $paramValue, $paramType);
+                    } else {
+                        throw new CentreonDbException(
+                            "Incorrect format for bindParam values, it must to be an array like ['value', PDO::PARAM_*]",
+                            ['bind_params' => $bindParams]
+                        );
+                    }
+                }
+                return $pdoStatement->execute();
+            }
+
+            return $pdoStatement->execute($bindParams);
+
+        } catch (PDOException $e) {
+            $message = "Error while executing the prepared query: {$e->getMessage()}";
+            $this->logSqlError($pdoStatement->queryString, $message);
+            throw new CentreonDbException(
+                $message,
+                [
+                    'query' => $pdoStatement->queryString,
+                    'pdo_error_code' => $e->getCode(),
+                    'pdo_error_infos' => $e->errorInfo,
+                ],
+                $e
+            );
+        } finally {
+            // here we restart CentreonDbStatement for the other requests
+            $this->setAttribute(PDO::ATTR_STATEMENT_CLASS, [
+                CentreonDBStatement::class,
+                [$this->log],
+            ]);
+        }
+    }
+
+    /**
+     * @param $query
+     * @param int $fetchMode
+     * @param array $fetchModeArgs
+     * @return PDOStatement|bool
+     * @throws CentreonDbException
+     */
+    public function executeQuery($query, int $fetchMode = PDO::FETCH_ASSOC, array $fetchModeArgs = []): PDOStatement|bool
+    {
+        if (empty($query)) {
+            throw new CentreonDbException(
+                'Error while executing query, query must not be empty',
+                [
+                    'query' => $query,
```

Comment:
$query is empty

Reply:
yes it's to show that $query is empty in the logs :) but ia can delete this if need